contrib/migrate/migrate.c

```diff
@@ -0,0 +1,897 @@
+
+  "alter table db_serial add column unique_name varchar first",
+  "delete from _db_attribute where class_of.class_name = 'db_serial' and rownum % 2 = 1 and attr_name <> 'unique_name'",
+
+  "alter table db_trigger add column unique_name varchar after owner",
+  "delete from _db_attribute where class_of.class_name = 'db_trigger' and rownum % 2 = 1 and attr_name <> 'unique_name'",
+
+  /* alter catalog to modify _db_index_key */
+  "alter table _db_index_key modify column func varchar(1023)",
+  "delete from _db_attribute where class_of.class_name = '_db_index_key' and rownum % 2 = 1",
+
+  /* alter catalog to add tables and views (_db_server, _db_synonym, db_server, db_synonym) */
+  "create table [_db_server] ( \
+  	[link_name] character varying(255) not null, \
+  	[host] character varying(255), \
+  	[port] integer, \
+  	[db_name] character varying(255), \
+  	[user_name] character varying(255), \
+  	[password] character varying(1073741823), \
+  	[properties] character varying(2048), \
+  	[owner] [db_user] not null, \
+  	[comment] character varying(1024), \
+  	constraint [pk__db_server_link_name_owner] primary key ([link_name], [owner]) \
+  ) dont_reuse_oid, collate utf8_bin",
+
+  "create view [db_server] ( \
+  	[link_name] character varying(255), \
+  	[host] character varying(255), \
+  	[port] integer, \
+  	[db_name] character varying(255), \
+  	[user_name] character varying(255), \
+  	[properties] character varying(2048), \
+  	[owner] character varying(256), \
+  	[comment] character varying(1024) \
+  ) as \
+	select \
+  	[ds].[link_name], \
+  	[ds].[host], \
+  	[ds].[port], \
+  	[ds].[db_name], \
+  	[ds].[user_name], \
+  	[ds].[properties], \
+  	[ds].[owner].[name], \
+  	[ds].[comment] \
+  from \
+  	[_db_server] [ds] \
+  where \
+  	current_user = 'DBA' \
+    or { [ds].[owner].[name] } subseteq ( \
+      select set {current_user} + coalesce (sum (set {[t].[g].[name]}), set { }) \
+      from [db_user] [u], table([groups]) as [t]([g]) \
+      where [u].[name] = current_user \
+    ) \
+    or { [ds] } subseteq ( \
+      select sum (set {[au].[class_of]}) \
+      from [_db_auth] [au] \
+      where {[au].[grantee].[name]} subseteq ( \
+                select set {current_user} + coalesce (sum (set {[t].[g].[name]}), set { }) \
+                from [db_user] [u], table([groups]) as [t]([g]) \
+                where [u].[name] = current_user \
+              ) \
+            and [au].[auth_type] = 'SELECT' \
+    )",
+
+  "CREATE TABLE [_db_synonym] ( \
+  	[unique_name] CHARACTER VARYING(255) NOT NULL, \
+  	[name] CHARACTER VARYING(255) NOT NULL, \
+  	[owner] [db_user] NOT NULL, \
+  	[is_public] INTEGER DEFAULT 0 NOT NULL, \
+  	[target_unique_name] CHARACTER VARYING(255) NOT NULL, \
+  	[target_name] CHARACTER VARYING(255) NOT NULL, \
+  	[target_owner] [db_user] NOT NULL, [comment] CHARACTER VARYING(2048), \
+  	CONSTRAINT [pk__db_synonym_unique_name] PRIMARY KEY  ([unique_name]), \
```

Comment:
It's trivial
Please reduce 2 spaces to 1 after "PRIMARY KEY".

Reply:
Fixed, thanks.